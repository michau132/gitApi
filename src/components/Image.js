import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageStyles = styled.img`
  object-fit: contain;
  max-height: 100%;
  max-width: 100%;
  height: 100%;
  width: 100%;
`;

const Image = ({ src, alt }) => (
  <ImageStyles src={src} alt={alt} />
);

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Image;
