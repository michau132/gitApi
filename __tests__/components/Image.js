import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Image from '../../src/components/Image';

describe('<Image /> component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      src: 'https://via.placeholder.com/150',
      alt: 'Beer',
    };
    wrapper = shallow(<Image {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders Image correctly', () => {
    expect(wrapper.prop('src')).toEqual(props.src);
    expect(wrapper.prop('alt')).toEqual(props.alt);
  });

});
