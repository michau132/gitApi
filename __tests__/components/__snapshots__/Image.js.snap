// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> component matching snapshot 1`] = `
<styled.img
  alt="Beer"
  src="https://via.placeholder.com/150"
/>
`;
